packages/effect/src/Schema.ts

```diff
@@ -9393,11 +9393,11 @@ export class BooleanFromString extends transform(
  * @category Config validations
  * @since 3.10.0
  */
-export const Config = <A>(name: string, schema: Schema<A, string>): config_.Config<A> => {
+export const Config = <A, B extends string>(name: string, schema: Schema<A, B>): config_.Config<A> => {
```

Comment:
I would 
- rename `B` to `I` for consistency
- remove the cast by using `ParseResult.decodeUnknownEither` instead of `decodeEither`
- and then use `ParseResult.TreeFormatter.formatIssueSync` instead of `ParseResult.TreeFormatter.formatErrorSync`

in pratice:

```ts
export const Config = <A, I extends string>(name: string, schema: Schema<A, I>): config_.Config<A> => {
  const decodeUnknownEither = ParseResult.decodeUnknownEither(schema)
  return config_.string(name).pipe(
    config_.mapOrFail((s) =>
      decodeUnknownEither(s).pipe(
        either_.mapLeft((error) => configError_.InvalidData([], ParseResult.TreeFormatter.formatIssueSync(error)))
      )
    )
  )
}
```